### PR TITLE
Fix perpetual use toughness as power of Walking bulwark

### LINKED ISSUE #10137: [[DMU]] Walking bulwarks ability isn't "until end of turn"
The ability doesn't go away at end of turn it still does 3 damage
[[Walking Bulwark]]

### PR DESCRIPTION
**Pull request by chesse20**
Resolves #10137 by making its toughness as attack until end of turn

### CODE REVIEW

File: Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java [hunk 1]
```diff
@@ -19,12 +19,20 @@ public class CombatDamageByToughnessEffect extends ContinuousEffectImpl {
...
 
     public CombatDamageByToughnessEffect(FilterCreaturePermanent filter, boolean onlyControlled) {
         super(Duration.WhileOnBattlefield, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
         this.filter = filter;
         this.onlyControlled = onlyControlled;
     }
+     
+    public CombatDamageByToughnessEffect(FilterCreaturePermanent filter, boolean onlyControlled, Duration duration) {
+        super(this.duration, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
+        this.filter = filter;
```
Comment (jbureau88) [inline Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java:32]: remove this here, you can't access this before super is called and also you don't put this when access a function parameter.

Comment (chesse20) [reply Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java:32]: ok, how should i overwrite the duration then with the one from the constructor?

Comment (chesse20) [reply Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java:32]: i think i have an idea actually

File: Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java [hunk 2]
```diff
@@ -19,12 +19,22 @@ public class CombatDamageByToughnessEffect extends ContinuousEffectImpl {
 
     private final FilterCreaturePermanent filter;
     private final boolean onlyControlled;
+    //private final Duration duration;
```
Comment (jbureau88) [inline Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java:22]: remove all commented code 👍

Comment (chesse20) [reply Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java]: yeap will do

**Comment by artemiswkearney** [comment]
Just tested this, seems to work
(tested:
- activation allows attacking for 3
- if something else allows attacking on a future turn after activation, damage is assigned based on power
- with a haste source in play, activating then bouncing then replaying, damage is assigned based on power
- with Arcades, the Strategist in play, activating doesn't clear Arcades's effect at end of turn
)

**Comment by xenohedron** [comment]
Will fix #10137

**Comment by chesse20** [comment]
ah i put the numbers in the wrong order

**Comment by xenohedron** [comment]
@theelk801 I think this one is good now

File: Mage.Sets/src/mage/cards/w/WalkingBulwark.java [hunk 3]
```diff
@@ -92,7 +92,7 @@ public boolean apply(Game game, Ability source) {
         game.addEffect(new CanAttackAsThoughItDidntHaveDefenderTargetEffect(Duration.EndOfTurn), source);
         FilterCreaturePermanent filter = new FilterCreaturePermanent();
         filter.add(new MageObjectReferencePredicate(permanent, game));
-        game.addEffect(new CombatDamageByToughnessEffect(filter, false), source);
+        game.addEffect(new CombatDamageByToughnessEffect(filter, false, Duration.EndOfTurn), source);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/w/WalkingBulwark.java:95]: you can simplify this with `.setDuration(Duration.EndOfTurn)`

File: Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java (hunk 1 above)
Comment (theelk801) [reply Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessEffect.java:32]: if you're gonna add a constructor for duration then only one should call the super

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/w/WalkingBulwark.java (hunk 3 above)
Comment (chesse20) [reply Mage.Sets/src/mage/cards/w/WalkingBulwark.java:95]: how would i implement a 2nd constructor with duration without calling the super? Sorry kinda rusty with java

Comment (theelk801) [reply Mage.Sets/src/mage/cards/w/WalkingBulwark.java:95]: you make a constructor that takes duration and calls the super and then another one which calls `this()` with the default duration choice

Comment (theelk801) [reply Mage.Sets/src/mage/cards/w/WalkingBulwark.java:95]: a good example would be `BoostTargetEffect`

**Review by theelk801 — APPROVED** [review]

**Pull request merged**